### DISCUSSION TITLE
Direnv .envrc support by mise

### DISCUSSION THREAD

**Discussion by jvizier** [jdx/mise#1988] [Ideas]
Hello,

Is there a way fir mise to read .envrc file (variables and/or functions) without direnv. (like mise able to read .tool-versions file)

We have a lots of .envrc and would like a transparent way to replace direnv by mise. Just to know if it's possible to do or maybe planned ?

Best regards,

**Comment by janeklb** [comment]
This would be a really nice quality-of-life improvement (though it would probably be a bit tedious to maintain). What I have started doing in my projects using `direnv`:

- extracting variables out of `.envrc` files into `.env` files
- add `dotenv .env` to the `.envrc` files (to ensure `direnv` continues to work)
- creating a `mise.toml` file (in each project) with
  ```toml
  [settings]
  env_file = ".env"
  ```

This is also pretty tedious, but it at least allows me to keep a single source of truth for the envars (via `.env` files) across `direnv` and `mise`

**Comment by jdx** [comment]
I think executing bash during mise initialization is best avoided for performance reasons. mise and direnv work completely differently—mise is declarative where direnv executes bash code for its configuration. This works alright for direnv which only runs on `cd` but mise runs every time the prompt is displayed. That doesn't mean it'd be impossible, but I don't have a strong appetite for taking it on or maintaining such functionality.
